### PR TITLE
Converted to azure function with timed trigger

### PR DESCRIPTION
**Pull request by Zindre17**

### CODE REVIEW

File: synk.spotify.csproj [hunk 1]
```diff
@@ -5,18 +5,30 @@
     <TargetFramework>net7.0</TargetFramework>
     <ImplicitUsings>enable</ImplicitUsings>
     <Nullable>enable</Nullable>
+    <AzureFunctionsVersion>v4</AzureFunctionsVersion>
   </PropertyGroup>
   
   <ItemGroup>
-    <None Update="appsettings.json">
+    <None Update="host.json">
```
Comment (adriansberg) [inline synk.spotify.csproj:12]: What potentially was in `application.json` needs to reside in `host.json` now if it is to be used?

File: host.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+    "version": "2.0",
+    "logging": {
+        "applicationInsights": {
```
Comment (adriansberg) [inline host.json:4]: I don't think we get Application Insights with the basic Function App I set up, can look into upgrading the Function App at some point.

File: PlayedTracksFetcher.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        this.tokenStore = tokenStore;
+        this.userStore = userStore;
+        this.musicStore = musicStore;
+        this.playedTracksStore = playedTracksStore;
+        this.tokenRefresher = tokenRefresher;
+        this.api = api;
+    }
+
+    [Function(nameof(PlayedTracksFetcher))]
+    public async Task Run([TimerTrigger("0 0 * * * *")] ScheduleInfo info)
```
Comment (adriansberg) [inline PlayedTracksFetcher.cs:31]: Can we move this trigger to a configuration somewhere so we don't need to update this code to update it? `0 0 * * * *` can be a default for that config however.

File: PlayedTracksFetcher.cs [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    {
+        this.tokenStore = tokenStore;
+        this.userStore = userStore;
+        this.musicStore = musicStore;
+        this.playedTracksStore = playedTracksStore;
+        this.tokenRefresher = tokenRefresher;
+        this.api = api;
+    }
+
+    [Function(nameof(PlayedTracksFetcher))]
```
Comment (adriansberg) [inline PlayedTracksFetcher.cs:30]: What will this result in in terms of URL? I.e. we have synk.azurewebsites.net, what will be after the slash?

**Review by adriansberg — APPROVED** [review]

File: synk.spotify.csproj (hunk 1 above)
Comment (Zindre17) [reply synk.spotify.csproj:12]: Nothing. But there is a local.settings.json that need have the values from appsettings when running locally. In deployed state we probably need to manually set these settings. 

Note: nesting is not supported in local.settings.json, so in order to "fake" an object, use __ to go one level in. 
Example: 
```
"Values": {
  "Spotify__ClientId" : "myclientid", 
  "Spotify__Secret": "mysecret"
}
```
is equivalent to 
```
"Spotify" : {
  "ClientId" : "myclientid",
  "Secret" : "mysecret"
}
```

File: host.json (hunk 2 above)
Comment (Zindre17) [reply host.json:4]: I see. This was just the defaults when generating a function through azure function commands.

File: PlayedTracksFetcher.cs (hunk 3 above)
Comment (Zindre17) [reply PlayedTracksFetcher.cs:31]: I don't think so. I think attribute arguments need to be compiletime-ready. But will have a look

File: PlayedTracksFetcher.cs (hunk 4 above)
Comment (Zindre17) [reply PlayedTracksFetcher.cs:30]: I am not sure it will be reachable through an endpoint when it has a TimerTrigger

**Pull request merged**